onecodex/cli.py

```diff
@@ -92,14 +103,36 @@ def scripts():
     pass
 
 
-scripts.add_command(filter_reads.cli, 'filter_reads')
+@scripts.command('filter_reads', help='Filter a FASTX file based on the taxonomic results from a CLASSIFICATION_ID')
```

Comment:
Do we have to move this in order for `login_required` to work? Ideally, we'd leave these and import them as the old code does. Otherwise, this file will become _very_ verbose as we add more and more scripts over time.

Reply:
The reason I did this originally is because I put the login_required decorator in cli.py, which imports scripts/filter_reads.py. In order to then access the login_required decorator from scripts/filter_reads.py, I'd have to circularly import login_required from cli.py.

Reply:
I was able to revert back to the scripts.add_command() style by moving login_required into auth.py and importing auth.login_required() in both filter_reads.py and cli.py.

Reply:
Though now, an Api() instance is returned in both cli.py and in auth.py, which doesn't seem very clean. Perhaps cli.login (which doesnt really do anything except call auth._login) should be refactored.

Reply:
If you want to see what I mean, checkout cli.login and note that it creates an Api() instance only to compare its e-mail address to the e-mail address returned by auth._login. I'm thinking that code can be moved into auth somewhere.